src/NuGetForUnity/Editor/InstalledPackagesManager.cs

```diff
@@ -0,0 +1,318 @@
+        internal static bool GetManuallyInstalledFlagFromConfiguration(string packageIdentifier)
+        {
+            var packageConfiguration = GetPackageConfigurationById(packageIdentifier);
+            return packageConfiguration == null || packageConfiguration.IsManuallyInstalled;
+        }
+
+        /// <summary>
+        ///     Sets the manually installed flag for the given package and stores the change in the <see cref="PackagesConfigFile" />.
+        /// </summary>
+        /// <param name="package">The package to mark as manually installed.</param>
+        internal static void SetManuallyInstalledFlag(INugetPackageIdentifier package)
+        {
+            PackagesConfigFile.SetManuallyInstalledFlag(package);
+            PackagesConfigFile.Save();
+        }
+
+        /// <summary>
+        ///     Invalidates the currently loaded 'packages.config' so it is reloaded when it is accessed the next time.
+        /// </summary>
+        internal static void ReloadPackagesConfig()
+        {
+            packagesConfigFile = null;
+        }
+
+        /// <summary>
+        ///     Removes the package from the 'packages.config' and from the installed packages dictionary (<see cref="InstalledPackages" />).
+        /// </summary>
+        /// <param name="package">The package to remove.</param>
+        internal static void RemovePackage(INugetPackage package)
+        {
+            // update the package.config file
+            if (PackagesConfigFile.RemovePackage(package))
+            {
+                PackagesConfigFile.Save();
+            }
+
+            InstalledPackagesDictionary.Remove(package.Id);
+        }
+
+        /// <summary>
+        ///     Adds the package to the 'packages.config' file.
+        /// </summary>
+        /// <param name="package">The package to add.</param>
+        internal static void AddPackageToConfig(INugetPackage package)
+        {
+            PackagesConfigFile.AddPackage(package);
+            PackagesConfigFile.Save();
+        }
+
+        /// <summary>
+        ///     Adds the package to the installed packages dictionary (<see cref="InstalledPackages" />).
+        /// </summary>
+        /// <param name="package">The package to add.</param>
+        internal static void AddPackageToInstalled(INugetPackage package)
+        {
+            InstalledPackagesDictionary.Add(package.Id, package);
+        }
+
+        /// <summary>
+        ///     Gets the package from the installed packages dictionary (<see cref="InstalledPackages" />) by its ID.
+        /// </summary>
+        /// <param name="id">The id of the package to search in the currently installed packages.</param>
+        /// <param name="package">The found installed package.</param>
+        /// <returns>True if a package was found.</returns>
+        internal static bool GetInstalledPackageById(string id, out INugetPackage package)
```

Comment:
I would name this method `TryGetById` since that naming is a standard for these kind of methods in C#.

Reply:
Nice name 👍